python/cucim/tests/unit/clara/test_tiff_read_region.py

```diff
@@ -0,0 +1,157 @@
+    tile_width, tile_height = cucim_img.resolutions['level_tile_sizes'][0]
+
+    # List of ((<start x>, <start y>), (<width>), (<height>))
+    region_list = [((-width - tile_width, -height - tile_height),
+                    (tile_width, tile_height)),  # left top (outside)
+                   ((width + tile_height, height + tile_height),
+                    (tile_width, tile_height)),  # right bottom (outside)
+                   ]
+
+    for (start_pos, size) in region_list:
+        cucim_arr = np.asarray(cucim_img.read_region(start_pos, size))
+        # All channel values should be zero, so we need to check that.
+        channel_value_count = np.apply_along_axis(np.count_nonzero, 2,
+                                                  cucim_arr)
+        count_all_zero = np.count_nonzero(channel_value_count == 0)
+        # All pixels would be zero.
+        assert count_all_zero == (tile_width * tile_height)
+
+
+def test_tiff_outside_of_resolution_level(testimg_tiff_stripe_4096x4096_256):
+    cucim_img = open_image_cucim(testimg_tiff_stripe_4096x4096_256)
+    with pytest.raises(ValueError, match=r"'level' should be less than "):
+        _ = cucim_img.read_region(level=-1)
+
+    with pytest.raises(ValueError, match=r"'level' should be less than "):
+        _ = cucim_img.read_region(level=7)
+
+
+def test_tiff_stripe_multiresolution(testimg_tiff_stripe_4096x4096_256):
+    cucim_img = open_image_cucim(testimg_tiff_stripe_4096x4096_256)
+
+    level_count = cucim_img.resolutions['level_count']
+    assert level_count == 6
+
+    start_pos, size = ((0, 0), (256, 256))
+    for level in range(level_count):
+        cucim_arr = np.asarray(cucim_img.read_region(start_pos, size, level))
+        # Not channel values are zero, so we need to check that.
+        channel_value_count = np.apply_along_axis(np.count_nonzero, 2,
+                                                  cucim_arr)
+        count_all_zero = np.count_nonzero(channel_value_count == 0)
+        img_size = cucim_img.resolutions['level_dimensions'][level]
+        # Only outside of the box is zero.
+        assert count_all_zero == 256 * 256 - (min(img_size[0], 256) *
+                                              min(img_size[1], 256))
+
+
+def test_array_interface_support(testimg_tiff_stripe_32x24_16_jpeg):
+    img = open_image_cucim(testimg_tiff_stripe_32x24_16_jpeg)
+    whole_img = img.read_region()
+    array_interface = whole_img.__array_interface__
+
+    # {'data': (45867600, False), 'strides': None,
+    #  'descr': [('', '|u1')], 'typestr': '|u1',
+    #  'shape': (24, 32, 3), 'version': 3}
+    assert array_interface['data'][0] is not None
+    assert not array_interface['data'][1]
+    assert array_interface['strides'] is None
+    assert array_interface['descr']
+    assert array_interface['shape'] == tuple(whole_img.shape)
+    assert array_interface['version'] == 3
+
+    def test_cuda_array_interface_support(testimg_tiff_stripe_32x24_16_jpeg):
```

Comment:
Doesn't this whole function need to be un-indented? why is it nested within `test_array_interface_support`?